cmd/oras/repository/tags.go

```diff
@@ -78,14 +78,36 @@ func showTags(opts showTagsOptions) error {
 	if err != nil {
 		return err
 	}
+	filter := ""
 	if opts.Reference != "" {
-		return fmt.Errorf("unexpected tag or digest %q found in repository reference %q", opts.Reference, opts.RawReference)
+		_, err := digest.Parse(opts.Reference)
+		if err == nil {
+			filter = opts.Reference
+		} else {
+			desc, err := finder.Resolve(ctx, opts.Reference)
+			if err != nil {
+				return err
+			}
+			filter = desc.Digest.String()
+		}
 	}
 	return finder.Tags(ctx, opts.last, func(tags []string) error {
 		for _, tag := range tags {
 			if opts.excludeDigestTag && isDigestTag(tag) {
 				continue
 			}
+			if filter != "" {
+				desc, err := finder.Resolve(ctx, tag)
```

Comment:
Scan and resolve every tag in the list will degrade the performance. Below are some experiments I made based on the current implementation:
```shell
oras repo tags docker.io/library/alpine | wc
time oras repo tags docker.io/library/alpine:latest
oras repo tags mcr.microsoft.com/cbl-mariner/distroless/minimal | wc
time oras repo tags mcr.microsoft.com/cbl-mariner/distroless/minimal:1.0
``` 

On my computer , it takes DockerHub 47.442 seconds to scan 129 tags, and MCR 19.192 seconds to scan 141 tags.


So we need to:
1) resolve descriptor and filter concurrently, also add `--concurrency` flag so user can use it to tune the behavior. This can be don in another PR.
2) print a warning log, reminding ORAS user how may tag need to be resolved and the command might take a long time to run
3) print experimental hint for this feature

Reply:
```
tlhowe@3c22fb41240e oras % oras repo tags --ca-file "${REGISTRY_CAFILE}" ${REGISTRY}/curated-packages/prometheus/prometheus:latest
Tag query by reference may take a while
v2
tlhowe@3c22fb41240e oras % oras repo tags --ca-file "${REGISTRY_CAFILE}" ${REGISTRY}/curated-packages/prometheus/prometheus:latest 2>err
v2
```

Reply:
https://github.com/oras-project/oras/issues/823 on the first issue. There is another issue ins oras-go to implement the method over there

Reply:
I think we can add warning and preview hint logs merge this PR first.